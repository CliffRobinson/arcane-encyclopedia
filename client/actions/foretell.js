export const FORETELL_ON = "FORETELL_ON";
export const FORETELL_OFF = "FORETELL_OFF";
export const FORETELL_TOGGLE = "FORETELL_TOGGLE";

export function foretellOn() {
    return {
        type: FORETELL_ON,
    };
}

export function foretellOff() {
    return {
        type: FORETELL_OFF,
    };
}

export function foretellToggle() {
    return {
        type: FORETELL_TOGGLE,
    };
}