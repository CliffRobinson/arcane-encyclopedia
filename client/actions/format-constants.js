export const STANDARD = "STANDARD";
export const HISTORIC = "HISTORIC";
export const M19 = "M19";
export const DOM = "DOM";
export const RIX = "RIX";
export const XLN = "XLN";
export const HOU = "HOU";
export const AKH = "AKH";
export const AER = "AER";
export const KLD = "KLD";
export const GRN = "GRN";
export const RNA = "RNA";
export const WAR = "WAR";
export const M20 = "M20";
export const ELD = "ELD";
export const THB = "THB";
export const IKO = "IKO"; //best guess for Ikoria set code
export const M21 = "M21";
export const AKR = "AKR";
export const ZNR = "ZNR"; //best guess for Zendikar Rising set code