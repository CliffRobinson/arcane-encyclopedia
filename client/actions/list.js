export const LIST_ON =  "LIST_ON";
export const LIST_OFF = "LIST_OFF";
export const LIST_TOGGLE = "LIST_TOGGLE";

export function listOn() {
    return {
        type: LIST_ON
    };
}

export function listOff() {
    return {
        type:LIST_OFF
    };
}

export function listToggle(){
    return {
        type: LIST_TOGGLE
    };
}