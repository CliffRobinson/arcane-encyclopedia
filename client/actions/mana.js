export const ADD_W = "ADD_W";
export const ADD_U = "ADD_U";
export const ADD_B = "ADD_B";
export const ADD_R = "ADD_R";
export const ADD_G = "ADD_G";
export const ADD_C = "ADD_C";

export const SUB_W = "SUB_W";
export const SUB_U = "SUB_U";
export const SUB_B = "SUB_B";
export const SUB_R = "SUB_R";
export const SUB_G = "SUB_G";
export const SUB_C = "SUB_C";

export const ADD_AZORIUS = "ADD_AZORIUS";
export const ADD_ORZHOV = "ADD_ORZHOV";
export const ADD_DIMIR = "ADD_DIMIR";
export const ADD_IZZET = "ADD_IZZET";
export const ADD_RAKDOS = "ADD_RAKDOS";
export const ADD_GOLGARI = "ADD_GOLGARI";
export const ADD_GRUUL = "ADD_GRUUL";
export const ADD_BOROS = "ADD_BOROS";
export const ADD_SELESNYA = "ADD_SELESNYA";
export const ADD_SIMIC = "ADD_SIMIC";

export const SUB_AZORIUS = "SUB_AZORIUS";
export const SUB_ORZHOV = "SUB_ORZHOV";
export const SUB_DIMIR = "SUB_DIMIR";
export const SUB_IZZET = "SUB_IZZET";
export const SUB_RAKDOS = "SUB_RAKDOS";
export const SUB_GOLGARI = "SUB_GOLGARI";
export const SUB_GRUUL = "SUB_GRUUL";
export const SUB_BOROS = "SUB_BOROS";
export const SUB_SELESNYA = "SUB_SELESNYA";
export const SUB_SIMIC = "SUB_SIMIC";

export const CLEAR_MANA = "CLEAR_MANA";

export function addW(){
    return {
        type: ADD_W
    };
}

export function addU(){
    return {
        type: ADD_U
    };
}

export function addB(){
    return {
        type: ADD_B
    };
}

export function addR(){
    return {
        type: ADD_R
    };
}

export function addG(){
    return {
        type: ADD_G
    };
}

export function addC(){
    return {
        type: ADD_C
    };
}

export function subW(){
    return {
        type: SUB_W
    };
}

export function subU(){
    return {
        type: SUB_U
    };
}

export function subB(){
    return {
        type: SUB_B
    };
}

export function subR(){
    return {
        type: SUB_R
    };
}

export function subG(){
    return {
        type: SUB_G
    };
}

export function subC(){
    return {
        type: SUB_C
    };
}


export function clearMana(){
    return {
        type: CLEAR_MANA
    };
}

export function addAzorius(){
    return {
        type: ADD_AZORIUS
    };
}

export function subAzorius(){
    return {
        type: SUB_AZORIUS
    };
}

// export function add[lower](){
//     return {
//         type: ADD_[upper]
//     };
// }

// export function sub[lower](){
//     return {
//         type: SUB_[upper]
//     };
// }

export function addOrzhov(){
    return {
        type: ADD_ORZHOV
    };
}

export function subOrzhov(){
    return {
        type: SUB_ORZHOV
    };
}

export function addDimir(){
    return {
        type: ADD_DIMIR
    };
}

export function subDimir(){
    return {
        type: SUB_DIMIR
    };
}

export function addIzzet(){
    return {
        type: ADD_IZZET
    };
}

export function subIzzet(){
    return {
        type: SUB_IZZET
    };
}

export function addRakdos(){
    return {
        type: ADD_RAKDOS
    };
}

export function subRakdos(){
    return {
        type: SUB_RAKDOS
    };
}

export function addGolgari(){
    return {
        type: ADD_GOLGARI
    };
}

export function subGolgari(){
    return {
        type: SUB_GOLGARI
    };
}

export function addGruul(){
    return {
        type: ADD_GRUUL
    };
}

export function subGruul(){
    return {
        type: SUB_GRUUL
    };
}

export function addBoros(){
    return {
        type: ADD_BOROS
    };
}

export function subBoros(){
    return {
        type: SUB_BOROS
    };
}

export function addSelesnya(){
    return {
        type: ADD_SELESNYA
    };
}

export function subSelesnya(){
    return {
        type: SUB_SELESNYA
    };
}

export function addSimic(){
    return {
        type: ADD_SIMIC
    };
}

export function subSimic(){
    return {
        type: SUB_SIMIC
    };
}