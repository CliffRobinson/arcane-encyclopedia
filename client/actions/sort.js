export const SORT_COLOR = "SORT_COLOR";     //As much as it pains me, it makes the most
export const SORT_CMC = "SORT_CMC";         //sense to spell colour in american across 
export const SORT_PRICE = "SORT_PRICE";     //the project.
export const SORT_NAME = "SORT_NAME";
export const SORT_COLLECTOR = "SORT_COLLECTOR";
export const SORT_RARITY = "SORT_RARITY";


export function sortColor() {
    return {
        type:SORT_COLOR
    };
}

export function sortCMC() {
    return {
        type:SORT_CMC
    };
}

export function sortPrice(){
    return {
        type:SORT_PRICE
    };
}

export function sortName(){
    return {
        type:SORT_NAME
    };
}

export function sortCollector(){
    return {
        type: SORT_COLLECTOR
    };
}

export function sortRarity() {
    return {
        type: SORT_RARITY
    };
}