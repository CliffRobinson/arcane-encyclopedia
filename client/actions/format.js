import * as constants from "./format-constants";

export function standard() {
    return {
        type: constants.STANDARD
    };
}

export function m19() {
    return {
        type: constants.M19
    };
}

export function dom() {
    return {
        type: constants.DOM
    };
}

export function rix() {
    return {
        type: constants.RIX
    };
}

export function xln() {
    return {
        type: constants.XLN
    };
}

export function hou() {
    return {
        type: constants.HOU
    };
}

export function akh() {
    return {
        type: constants.AKH
    };
}

export function aer() {
    return {
        type: constants.AER
    };
}

export function kld() {
    return {
        type: constants.KLD
    };
}

export function grn() {
    return {
        type: constants.GRN
    };
}

export function rna() {
    return {
        type: constants.RNA
    };
}

export function war() {
    return {
        type: constants.WAR
    };
}

export function m20() {
    return {
        type: constants.M20
    };
}

export function eld() {
    return {
        type: constants.ELD
    };
}

export function thb() {
    return {
        type: constants.THB
    };
}

export function ikr() {
    return {
        type: constants.IKR
    };
}

export function zkr() {
    return {
        type: constants.ZKR
    };
}