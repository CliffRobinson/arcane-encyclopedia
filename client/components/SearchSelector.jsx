import React from "react";
import { connect } from "react-redux";
import request from "superagent";
//import {createQuery, getCardsFromScryfall} from "./componentFunctions";

import * as formatActions from "../actions/format";
import * as sortActions from "../actions/sort";
import {addCards, clearCards} from "../actions/cards";
import {tricksToggle} from "../actions/onlyTricks";
import {landsToggle} from "../actions/filterLands";

import fakeCards from "../../tests/testData.json";

class SearchSelector extends React.Component {
    constructor(props) {
        super(props);
        this.searchHandleChange = this.searchHandleChange.bind(this);
        this.sortHandleChange = this.sortHandleChange.bind(this);
        //this.createQuery = this.createQuery.bind(this);
        //this.getCardsFromScryfall = this.getCardsFromScryfall.bind(this);
        this.getFakes = this.getFakes.bind(this);

    }


    render() {
        const size = "is-small";
        return (
            <div className="searchSelector" > {/*navbar-item*/}
                <select className= {`select ${size}`} onChange={this.searchHandleChange}> {/*values of options correspond to the action to dispatch*/}
                    <option value="standard"> Standard</option>
                    <option value="aer"> Aether Revolt Limited</option>
                    <option value="hou"> Hour of Devastation Limited</option>
                    <option value="rix"> Rivals of Ixalan Limited </option>
                    <option value="dom"> Dominaria Limited</option>
                    <option value="m19"> M19 Limited</option>
                </select>
                <button className={`button ${size}`} onClick={() => this.createQuery()/*createQuery.call(this)*/} > Get Cards </button>
                <button className={`button ${size}`} onClick={()=> this.props.dispatch(clearCards())}> Clear Cards </button>

                <select className={`select ${size}`} onChange={this.sortHandleChange}> {/*values of options correspond to the action to dispatch*/} 
                    <option value ="sortName" > Sort by Name </option>
                    <option value ="sortColor" > Sort by Color </option>
                    <option value ="sortCMC" > Sort by CMC </option>
                    <option value ="sortPrice" > Sort by Price </option>
                    <option value ="sortCollector" > Sort by Collector </option>
                    <option value="sortRarity"> Sort by Rarity </option>
                </select>

                <input type="checkbox" onClick={()=> this.props.dispatch(tricksToggle())} /> Only show tricks 
                <input type="checkbox" onChange={()=> this.props.dispatch(landsToggle())} defaultChecked={true} /> Exclude Lands

                <button className={`button ${size}`} onClick={this.getFakes}> Get Fakes </button>
            </ div>
        );
    }

    getFakes() {
        this.props.dispatch(clearCards());
        let fakes = fakeCards.data;
        this.props.dispatch(addCards(fakes));
    }

    createQuery(){
        this.props.dispatch(clearCards());
        let queryString = `https://api.scryfall.com/cards/search?q=${this.props.format}`;
        this.getCardsFromScryfall(queryString);
    }

    getCardsFromScryfall(queryString) {
        
        request.get(queryString)
            .then( (res) =>{
                this.props.dispatch(addCards(res.body.data));

                if (res.body.has_more)  {
                    this.getCardsFromScryfall(res.body.next_page);
                }
            });
    }

    searchHandleChange(e) {
        this.props.dispatch(formatActions[e.target.value]());
    }

    sortHandleChange(e){
        this.props.dispatch(sortActions[e.target.value]());
    }

}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps)(SearchSelector);