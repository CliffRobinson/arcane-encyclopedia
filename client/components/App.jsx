import React from "react";
import Cards from "./Cards";
import ManaSelector from "./ManaSelector";
import SearchSelector from "./SearchSelector";
import FilterBar from "./FilterBar";


const App = () => {
    return (
        <div className="App" >
            <div className="topTwo">
                <SearchSelector />
                <FilterBar />
            </div>

            <div className="flexContain">
                <div className="main">
                    <Cards />
                </div>
                <div className="sideColumn">
                    <ManaSelector />
                </div>
                
            </div>
        </div>
    );
};

export default App;

