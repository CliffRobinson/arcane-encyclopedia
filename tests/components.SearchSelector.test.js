import {SearchSelector} from "../client/components/SearchSelector";
import {clearCards, addCards} from "../client/actions/cards";
import {data as fakeCards}  from "./testData.json";

test("getFakes dispatches correct actions", ()=> {
    //Arrange
    const mockDispatch = jest.fn();
    const testSelector = new SearchSelector({
        dispatch:mockDispatch
    });
    //Act
    testSelector.getFakes();
    //Assert
    expect(mockDispatch.mock.calls.length).toBe(2);
    expect(mockDispatch.mock.calls[0][0]).toEqual(clearCards());
    expect(mockDispatch.mock.calls[1][0]).toEqual(addCards(fakeCards));
});

test("createQuery dispatches correct actions", ()=> {
    //Arrange
    const mockDispatch = jest.fn();
    const formatString = "f:standard";
    const testSelector = new SearchSelector({
        dispatch:mockDispatch,
        format: formatString
    });
    const mockGCFS = jest.fn();
    testSelector.getCardsFromScryfall = mockGCFS;
    //Act
    testSelector.createQuery();
    //Assert
    expect(mockDispatch.mock.calls.length).toBe(1);
    expect(mockDispatch.mock.calls[0][0]).toEqual(clearCards());
    expect(mockGCFS.mock.calls[0][0]).toEqual(`https://api.scryfall.com/cards/search?q=${formatString}`);
});

test("getCardsFromScryfall dispatches res.body.data when res.body.has_more == false", ()=> {
    //Arrange
    const mockGet = jest.fn();
    mockGet.mockReturnValue({
        body: {
            has_more:false,
            data:fakeCards
        }
    }); 
    const mockRequest = {
        get:mockGet
    };
    const mockDispatch = jest.fn();
    const testSelector = new SearchSelector({
        dispatch:mockDispatch,
    });
    testSelector.request = mockRequest;
    //Act
    testSelector.getCardsFromScryfall();
    //Assert
    expect(mockGet.mock.calls.length).toBe(1);
});