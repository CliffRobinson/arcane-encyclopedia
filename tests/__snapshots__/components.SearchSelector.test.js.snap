// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchSelector matches snapshot 1`] = `
<div
  className="searchSelector"
>
   
  <select
    className="select is-small"
    onChange={[Function]}
  >
    <option
      value="STANDARD"
    >
      STANDARD
    </option>
    <option
      value="HISTORIC"
    >
      HISTORIC
    </option>
    <option
      value="M19"
    >
      M19
    </option>
    <option
      value="DOM"
    >
      DOM
    </option>
    <option
      value="RIX"
    >
      RIX
    </option>
    <option
      value="XLN"
    >
      XLN
    </option>
    <option
      value="HOU"
    >
      HOU
    </option>
    <option
      value="AKH"
    >
      AKH
    </option>
    <option
      value="AER"
    >
      AER
    </option>
    <option
      value="KLD"
    >
      KLD
    </option>
    <option
      value="GRN"
    >
      GRN
    </option>
    <option
      value="RNA"
    >
      RNA
    </option>
    <option
      value="WAR"
    >
      WAR
    </option>
    <option
      value="M20"
    >
      M20
    </option>
    <option
      value="ELD"
    >
      ELD
    </option>
    <option
      value="THB"
    >
      THB
    </option>
    <option
      value="IKO"
    >
      IKO
    </option>
    <option
      value="M21"
    >
      M21
    </option>
    <option
      value="AKR"
    >
      AKR
    </option>
    <option
      value="ZKR"
    >
      ZKR
    </option>
  </select>
  <button
    className="button is-small"
    onClick={[Function]}
  >
     Get Cards 
  </button>
  <button
    className="button is-small"
    onClick={[Function]}
  >
     Clear Cards 
  </button>
  <select
    className="select is-small"
    onChange={[Function]}
  >
    <option
      value="sortName"
    >
       Sort by Name 
    </option>
    <option
      value="sortColor"
    >
       Sort by Color 
    </option>
    <option
      value="sortCMC"
    >
       Sort by CMC 
    </option>
    <option
      value="sortPrice"
    >
       Sort by Price 
    </option>
    <option
      value="sortCollector"
    >
       Sort by Collector 
    </option>
    <option
      value="sortRarity"
    >
       Sort by Rarity 
    </option>
  </select>
  <input
    onClick={[Function]}
    type="checkbox"
  />
   Only show tricks
  <input
    defaultChecked={true}
    onChange={[Function]}
    type="checkbox"
  />
   Exclude Lands
</div>
`;
